pkgs/development/compilers/scopes/default.nix

```diff
@@ -0,0 +1,63 @@
+{ stdenv
+, fetchFromBitbucket
+, llvm
+, clang
+, clang-unwrapped
+, genie
+, spirv-tools
+, spirv-cross
+, libc-hdrs
+, clang-hdrs
+}:
+stdenv.mkDerivation rec {
+  pname = "scopes";
+  version = "unstable-2019-05-09";
+  src = fetchFromBitbucket {
+    owner = "duangle";
+    repo = "scopes";
+    rev = "3c5fd78";
+    sha256 = "1im9lm0j4j4d797mwn97zfsfh91rls1arw52filzdi3wlq4zgxfi";
+  };
+  nativeBuildInputs = [ genie spirv-tools spirv-cross ];
+  buildInputs = [ llvm clang-unwrapped ];
+  postPatch = ''
+    substituteInPlace genie.lua \
```

Comment:
Patching like this will prove to be problematic as if any of the substitutions fail it's going to be very silent about it.

Using a patch with `substituteAll` where you can put environment variables in the patch to replace them with nix store paths would be much better.

See https://github.com/NixOS/nixpkgs/blob/9e6740c7ad610abba13a40f02d5eac93874b8bc8/pkgs/development/libraries/flatpak/default.nix#L19 for example. Note that the variables must be valid in bash.

Reply:
Done.